### DISCUSSION TITLE
make tag creation optional

### DISCUSSION THREAD

**Discussion by TheKangaroo** [commitizen-tools/commitizen#730] [Q&A] [answered]
### Description

It would be great for our workflow to make it possible to opt out via config option of the creation of a git tag on `cz bump`.

### Possible Solution

we could add a config option like `disable_tag_creation` (which defaults to false) to make it possible to optional disable the creation of the git tag.

### Additional context

We just started to use commitizen so maybe our workflow isn't the best right now.

Our GitLab rules forbid to push on main and we don't want to change that. So we needed to find a way to create releases with commitizen without pushing to main. We came up with the following workflow:
* create new branch localy from main
* run `cz bump` (with changelog creation) (which create a local tag on that feature branch)
* push a merge request (just the commit without the tag) to the upstream repo
* have a teammate review and approve the merge request
* merge the merge request
* a ci pipeline will run on every merge to master with commit message "bump: version.*" and create a gitlab release for it which also creates a git tag

But this means we end up with tags on different commits locally than in our upstream repo.
We can clean things up by running
```
git tag -l | xargs git tag -d
git fetch --tags
```
but it would be good to just don't mess up things in the first place.

As I said, we're new to commitizen, so maybe there is an easier way to achive this

**Comment by bruno-fs** [comment]
This workflow would work better for my team as well

**Comment by woile** [comment]
Can you expand on your workflow @bruno-fs ?

**Comment by TheKangaroo** [comment]
I have to admit, we switched to release-it with conventional-changelog and bumper plugin for our release workflow, which made our workflow work without any hacks.

**Comment by woile** [comment, marked as answer]
That's great @TheKangaroo !  Glad to hear you found a solution!

For reference in commitizen you could do:

```sh
cz bump --files-only --yes --changelog
git commit -am "bump: release $(cz version --project)"
```

And later on to create the tag:

```
git tag $(cz version --project)
```

Commitizen works well with git, and where commitizen is available you know `git` will be available, this gives you a lot of flexibility.

Maybe we could add this as a tutorial? 🤔

**Reply by chezhj** [reply]
Searched for hours to find this, that after a "--files-only" a manual git tag is needed. I was looking for a flag in "cz bump" Add: this is not complete if you use tag_format. I guess there is no way to get the tag?
cz version --project --tag
